sklearn/tests/test_naive_bayes.py

```diff
@@ -698,6 +700,51 @@ def test_categoricalnb():
         clf = CategoricalNB(alpha=1, fit_prior=False)
         clf.fit(X, y, sample_weight=sample_weight)
         assert_array_equal(clf.predict(np.array([[0, 0]])), np.array([2]))
+        assert_array_equal(clf.n_categories_, np.array([2, 2]))
+
+
+def test_categoricalnb_with_min_categories():
+    # None as n_categories is checked by default in test_categoricalnb
+
+    X_n_categories = np.array([[0, 0], [0, 1], [0, 0], [1, 1]])
+    y_n_categories = np.array([1, 1, 2, 2])
+
+    # Check min_categories with int
+    clf = CategoricalNB(alpha=1, fit_prior=False, min_categories=3)
+    clf.fit(X_n_categories, y_n_categories)
+    X1_count, X2_count = clf.category_count_
+    assert_array_equal(X1_count, np.array([[2, 0, 0], [1, 1, 0]]))
+    assert_array_equal(X2_count, np.array([[1, 1, 0], [1, 1, 0]]))
+    data_with_unseen_feature_2 = np.array([[0, 2]])
+    predictions = clf.predict(data_with_unseen_feature_2)
+    assert_array_equal(predictions, np.array([1]))
+    assert_array_equal(clf.n_categories_, np.array([3, 3]))
+
+    # Check min_categories with list
+    clf = CategoricalNB(alpha=1, fit_prior=False, min_categories=[3, 4])
+    clf.fit(X_n_categories, y_n_categories)
+    X1_count, X2_count = clf.category_count_
+    assert_array_equal(X1_count, np.array([[2, 0, 0], [1, 1, 0]]))
+    assert_array_equal(X2_count, np.array([[1, 1, 0, 0], [1, 1, 0, 0]]))
+    data_with_unseen_feature_3 = np.array([[0, 3]])
+    predictions = clf.predict(data_with_unseen_feature_3)
+    assert_array_equal(predictions, np.array([1]))
+    assert_array_equal(clf.n_categories_, np.array([3, 4]))
+
+    # check min_categories with min less than actual
+    clf = CategoricalNB(alpha=1, fit_prior=False, min_categories=1)
+    clf.fit(X_n_categories, y_n_categories)
+    assert_array_equal(clf.n_categories_, np.array([2, 2]))
+
+    # does not accept string
+    clf = CategoricalNB(alpha=1, fit_prior=False, min_categories='bad_arg')
+    assert_raises(ValueError, clf.fit, X_n_categories, y_n_categories)
```

Comment:
You should use pytest instead of `assert_raises` (we are removing those):

```python
with pytest.raises(ValueError, match=err_msg):
    CategoricalNB(....).fit(X, y)
```

`err_msg` will be a partial string which will correspond to the error (partially) message.
Ensure that we match the error message each time to be sure that we are raising the error that we expect.

Reply:
Sounds good, done!